include/nbl/video/SPhysicalDeviceFeatures.h

```diff
@@ -0,0 +1,102 @@
+#ifndef _NBL_VIDEO_S_PHYSICAL_DEVICE_FEATURES_H_INCLUDED_
+#define _NBL_VIDEO_S_PHYSICAL_DEVICE_FEATURES_H_INCLUDED_
+
+namespace nbl::video
+{
+
+struct SPhysicalDeviceFeatures
+{
+    /* Vulkan Core 1.0 */
+    bool robustBufferAccess = false;
+    //VkBool32    fullDrawIndexUint32;
+    bool imageCubeArray = false;
+    //VkBool32    independentBlend;
+    bool geometryShader    = false;
+    //VkBool32    tessellationShader;
```

Comment:
these we want to expose under GL definitely
https://github.com/Devsh-Graphics-Programming/Nabla/pull/348#discussion_r867400005

Reply:
in GL 4.4 (our minium) these are always true, but in GLES (even 3.1 out minimum( and Vulkan this stuff is optional

Reply:
`fullDrawIndexUint32`: for OpenGL I'll set to true but I don't know how to check for gles (`isGLES is true`)
`imageCubeArray`: already implemented -> `= true; //we require OES_texture_cube_map_array on GLES`
`independentBlend`: for OpenGL I'll set to true but I don't know how to check for gles 
`geometryShader`: already implemented -> checking `m_glfeatures.NBL_ARB_geometry_shader4`
`tessellationShader`: will implement -> by checking `NBL_ARB_tessellation_shader`

Reply:
`fullDrawIndexUint32` check if `GL_MAX_ELEMENT_INDEX` is equal to `2^32-1`

> `geometryShader`: already implemented -> checking `m_glfeatures.NBL_ARB_geometry_shader4`
> `tessellationShader`: will implement -> by checking `NBL_ARB_tessellation_shader`

both are core in desktop GL

in GLES you need to check more than 1 extension I think.

Reply:
> in GLES you need to check more than 1 extension I think.

for tessellation and geometry you have 4 different ways (in decreasing order of preference) you can get access to the feature:
- GLES Version >= 3.2 (no suffix)
- OES extension (OES is like ARB but for GLES)
- EXT extension
- INTEL extension 

`ARB_geometry_shader4` is an dekstop GL extension (which we don't need to check)